Model/Sync/Checkout/Data.php

```diff
@@ -206,13 +206,13 @@ public function getDataBeforeChange()
     /**
      * Prepare address data
      * @param Quote $quote
-     * @param string $type
+     * @param bool $isBillingAddress
      * @return array<mixed>
      * @throws LocalizedException
      */
-    public function prepareAddress(Quote $quote, string $type)
+    public function prepareAddress(Quote $quote, $isBillingAddress)
```

Comment:
Passing flags is not considered a great practice.
Let's have a separate method for shipping address and use it.

Reply:
Sure, separated to `prepareBillingAddress` and `prepareShippingAddress`.